Sources/PackageGraph/Pubgrub/PubgrubDependencyResolver.swift

```diff
@@ -1410,10 +1416,10 @@ fileprivate extension PackageRequirement {
 
 fileprivate extension DependencyResolutionNode {
     var nameForDiagnostics: String {
-        if let product = specificProduct {
-            return "\(package.name)[\(product)]"
+        if let product = specificProduct, product != package.name{
+            return "'\(package.name)::\(product)'"
```

Comment:
I think the `::` form a lot more than the bracket form, which never made sense to me.